Engine/DataFeeds/SubscriptionCollection.cs

```diff
@@ -117,17 +126,70 @@ IEnumerator IEnumerable.GetEnumerator()
             return GetEnumerator();
         }
 
+        /// <summary>
+        /// Helper method to validate a configuration to be included in the fill forward calculation
+        /// </summary>
+        private static bool ValidateFillForwardResolution(SubscriptionDataConfig configuration)
+        {
+            return !configuration.IsInternalFeed && configuration.Resolution != Resolution.Tick;
+        }
+        /// <summary>
+        /// Gets and updates the fill forward resolution by checking specified subscription configurations and
+        /// selecting the smallest resoluton not equal to tick
+        /// </summary>
+        private void UpdateFillForwardResolution(bool adding, SubscriptionDataConfig configuration)
+        {
+            // Due to performance implications let's be jealous in updating the _fillForwardResolution
+            if (ValidateFillForwardResolution(configuration) &&
+                (
+                    (adding  // we are adding
+                    && configuration.Resolution.ToTimeSpan() != _fillForwardResolution.Value) // check if the new timeSpan is different
```

Comment:
I think we can even do a _little_ better here. First, let's use `configuration.Increment` which is the time span equivalent of resolution as well, second I think we only need to update the fill forward resolution if the added subscription's increment is smaller.

Reply:
I'll use `configuration.Increment` 👍 
Right, I thought the same. Problem is, the _default_ fill forward enumerator resolution might be smaller than the users resolution. So we should add logic to detect when the `_subscriptions` collection is empty for overwriting the _smaller_ logic, and that stopped me from doing so since its an operation with awful performance, what do you think?

Reply:
good point, I think the cost savings probably aren't worth the added complexity.